src/appsignal/config.py

```diff
@@ -3,16 +3,7 @@
 import os
 import platform
 import tempfile
-from typing import (
-    TYPE_CHECKING,
-    List,
-    Literal,
-    Optional,
-    TypedDict,
-    Union,
-    cast,
-    get_args,
-)
+from typing import TYPE_CHECKING, List, Literal, TypedDict, cast, get_args
```

Comment:
Asking in order to better understand how type checking works: I see that, while most annotations have been changed to use `list[...]`, we still import `List` here, so that we can use it in expressions like `cast(List[DefaultInstrumentation]`.

Is it that we can't use `cast(list[DefaultInstrumentation])` because the call to `cast` happens in the run-time code, not in annotations (even if it's a no-op at run-time) and `list[...]` is not valid syntax at run-time in (older?) versions of Python?

And, because you've added `from __future__ import annotations`, the Python interpreter no longer checks what's going on within annotations, which allows us to use the newer `list[...]` form there. Did I get that right?

Reply:
Precisely! That's exactly what happens. Other examples of type annotations evaluated at runtime include type aliases, `bind` argument for `TypeVar`, base classes like `Generic` and their arguments, and pydantic models.